File: golang/121_best_time_to_buy_and_sell_stock.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func maxProfit(prices []int) int {
	ans, l, r := 0, 0, 0
	for r < len(prices) {
		if prices[l] <= prices[r] {
			diff := prices[r] - prices[l]
			if diff > ans {
				ans = diff
			}
			r++
		} else {
			l++
		}
	}
	return ans
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	line := scanner.Text()
	splittedInput := strings.Split(line, ",")
	var nums []int
	for i := range splittedInput {
		num, err := strconv.Atoi(splittedInput[i])
		if err == nil {
			nums = append(nums, num)
		}
	}
	fmt.Println(maxProfit(nums))
}

File: python/121_best_time_to_buy_and_sell_stock.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        l, r = 0, 0
        while r < len(prices):
            if prices[l] <= prices[r]:
                ans = max(ans, prices[r]-prices[l])
                r += 1
            else:
                l += 1
        return ans


sol = Solution()
inp = [int(x) for x in input().split(",")]
print(sol.maxProfit(inp))